theseus/core/cost_function.py

```diff
@@ -128,8 +115,8 @@ def __init__(
             raise ValueError(
                 "AutodiffCostFunction must receive at least one optimization variable."
             )
-        _register_vars_in_list(self, optim_vars, is_optim=True)
-        _register_vars_in_list(self, aux_vars, is_optim=False)
+        self.register_vars(optim_vars, optim_vars=True)
```

Comment:
`is_optim_vars`? The current name could lead to some error/confusion.

Reply:
Changed.